File: Source/OregonTilth.Models/DataTransferObjects/Workbook/Forms/TimeStudy/TimeStudiesUpsertDto.cs
using System.ComponentModel.DataAnnotations;
using System.Dynamic;

namespace OregonTilth.Models.DataTransferObjects
{
    public class TimeStudiesUpsertDto
    {
        public TimeStudyUpsertDto[] TimeStudies { get; set; }
        public int WorkbookID { get; set; }
        public int? FieldStandardTimeID { get; set; }
        public int? HarvestPostHarvestStandardTimeID { get; set; }
    }
}

File: Source/OregonTilth.Web/src/app/shared/models/forms/time-studies/time-studies-upsert-dto.ts
import { TimeStudyDto } from "../../generated/time-study-dto";
import { FieldStandardTimeSummaryDto } from "../field-standard-times/field-standard-time-summary-dto";

export class TimeStudiesUpsertDto {
    TimeStudies: TimeStudyUpsertDto[];
    WorkbookID: number;
    FieldStandardTimeID: number;
    HarvestPostHarvestStandardTimeID: number;
    TransplantProductionStandardTimeID: number;
    constructor(obj?: any) {
        Object.assign(this, obj);
    }
}

export class TimeStudyUpsertDto {
    TimeStudyID: number;
    FieldStandardTimeID: number;
    HarvestPostHarvestStandardTimeID: number;
    TransplantProductionStandardTimeID: number;
    Notes: string;
    Units: number;
    Duration: number;
    WorkbookID: number;
    constructor(obj?: any) {
        Object.assign(this, obj);
    }
}

export class TimeStudySummaryDto {
    TimeStudyID: number;
    FieldStandardTimeID: number;
    HarvestPostHarvestStandardTimeID: number;
    TransplantProductionStandardTimeID: number;
    Notes: string;
    Units: number;
    Duration: number;
    WorkbookID: number;
    constructor(obj?: any) {
        Object.assign(this, obj);
    }
}